src/lib/validators.js

```diff
@@ -22,10 +22,10 @@ const validators = {
    */
   allow: function(def, value, key, errors) {
     const type = 'allow'
-    const sub = def.allow
-    if (Array.isArray(def.allow) && def.allow.indexOf(value) === -1) {
+    const sub = typeof def.allow === 'object' && !Array.isArray(def.allow) ? Object.keys(def.allow) : def.allow
+    if (Array.isArray(sub) && sub.indexOf(value) === -1) {
       errors.push({ type, sub, key, value, message: `Value '${value}' is not allowed` })
-    } else if (!Array.isArray(def.allow) && def.allow !== value) {
+    } else if (!Array.isArray(sub) && sub !== value) {
```

Comment:
`Array.isArray()` is called 3 times here... It's probably lightweight so this might be a micro-optimization, but might be worth it to store the result of one call and just refer to that.